esss_fix_format/cli.py

```diff
@@ -157,6 +159,22 @@ def _process_file(filename, check, format_code):
         click.secho(click.format_filename(filename) + ': Unknown file type', fg='white')
         return changed, errors, formatter
 
+    if is_cpp(filename):
+        with io.open(filename, 'rb') as f:
+            content_bytes = f.read()
+            content = content_bytes.decode()
+            non_ascii = re.sub('[ -~]', '', content).strip()
+            use_bom = len(non_ascii) > 0
+            if use_bom and not content_bytes.startswith(codecs.BOM_UTF8):
```

Comment:
I think other encodings are also valid (utf 16?).

Reply:
Hmmm.. Indeed, that will fail for other encoding besides UTF-8. Will fix it.

Reply:
But doesn't the issue specifically asked to check against cpp files, which contains non-ascii characters, to have "UTF-8 encoding with BOM"?